src/hooks/use-query.tsx

```diff
@@ -0,0 +1,28 @@
+import {
+  useQuery as useReactQuery,
+  QueryKey,
+  UseQueryOptions
+} from '@tanstack/react-query'
+import { ErrorResponse } from '~/types'
+
+interface UseQueryProps<TData, TError, TSelect> {
+  queryKey: QueryKey
+  queryFn: () => Promise<TData>
+  options?: UseQueryOptions<TData, TError, TSelect>
+}
+
+const useQuery = <TData, TError = ErrorResponse, TSelect = TData>({
+  queryKey,
+  queryFn,
```

Comment:
does `queryFn` stand for queryFunction? If that, please rename it to `queryFunction`, as it is more readable

Reply:
`queryFn` is a name from `@tanstack/react-query` library, I think this name is better here for consistency with [@tanstack/react-query API](https://tanstack.com/query/latest/docs/framework/react/reference/useQuery)

Reply:
okay, got it, then it’s better to leave it like that